app/js/components/mixins/ModalEffects.react.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+
+let ModalEffects = InnerComponent => class extends React.Component {
```

Comment:
Hopefully, they'd become more popular than they currently are by the time new developer come in 😄 
But it's actually a straight forward implementation once you wrap your head around it. Thanks for the voice of caution though. We should write code with the future generation in mind.